__tests__/integration/hathorwallet_facade.test.js

```diff
@@ -30,6 +30,158 @@ import MemoryStore from '../../src/memory_store'
+    jest.spyOn(hWallet, 'getFullTxById').mockResolvedValue({
+      success: true,
+      tx: {
+        ...tx1,
+        // impossible token_data
+        inputs: [{ ...tx1.inputs[0], token_data: -1 }],
+      },
+    });
+    await expect(hWallet.getTxById(tx1.hash)).rejects.toThrowError('Token undefined not found in tokens list');
+    jest.spyOn(hWallet, 'getFullTxById').mockRestore();
+
+    // thorw error if token not found in tx
+    jest.spyOn(hWallet, 'getTxBalance').mockResolvedValue({
+      'unknown-token': 10,
+    });
+    await expect(hWallet.getTxById(tx1.hash)).rejects.toThrowError('Token unknown-token not found in tx');
+    jest.spyOn(hWallet, 'getTxBalance').mockRestore();
+  });
+
+  it('should throw an error tx id is invalid', async () => {
+    const hWallet = await generateWalletHelper();
+    await expect(hWallet.getTxById('invalid-tx-hash')).rejects.toThrowError('Invalid transaction invalid-tx-hash');
+  });
+
+  it('should get the balance for a custom token', async () => {
+    const hWallet = await generateWalletHelper();
+
+    // Test case: non-existent token
+    const emptyBalance = await hWallet.getBalance(fakeTokenUid);
+    // Assert that only one balance is returned
+    expect(emptyBalance).toHaveLength(1);
+    // Assert the balance is zero
+    expect(emptyBalance[0]).toMatchObject({
+      token: { id: fakeTokenUid },
+      balance: { unlocked: 0, locked: 0 },
+      transactions: 0,
+    });
+
+    // Test case: custom token with funds
+    const address = hWallet.getAddressAtIndex(0);
+    // Inject 10 HTR into the wallet
+    await GenesisWalletHelper.injectFunds(address, 10);
+    // Generate a random amount of new tokens
+    const newTokenAmount = getRandomInt(1000, 10);
+    // Create a new custom token with the generated amount
+    const { hash: tokenUid } = await createTokenHelper(
+      hWallet,
+      'BalanceToken',
+      'BAT',
+      newTokenAmount,
+    );
+    // Get the balance of the new token
+    const tknBalance = await hWallet.getBalance(tokenUid);
+    // Assert that only one balance is returned
+    expect(tknBalance).toHaveLength(1);
+    // Assert the balance is equal to the amount generated
+    expect(tknBalance[0]).toMatchObject({
+      token: { id: tokenUid },
+      balance: { unlocked: newTokenAmount, locked: 0 },
+      transactions: expect.any(Number),
+      /**
+       * TODO: The amount of transactions is often 8 but should be 1. Ref #397
+       * @see https://github.com/HathorNetwork/hathor-wallet-lib/issues/397
+       */
+      // transactions: 1, 
+    });
+    // Get balance for the token creation transaction
+    const result = await hWallet.getTxById(tokenUid);
+    expect(result.success).toStrictEqual(true);
+    expect(result.txTokens).toHaveLength(2);
+    expect(result.txTokens).toEqual(expect.arrayContaining([
+      expect.objectContaining({
+        tokenId: HATHOR_TOKEN_CONFIG.uid,
+        balance: expect.any(Number),
```

Comment:
💬 It would be nice to have the jest-extended lib here and use the asymmetric matcher [`toBeNegative`](https://jest-extended.jestcommunity.dev/docs/matchers/number/#tobenegative).